internal/controllers/clusterextension_controller.go

```diff
@@ -203,21 +203,34 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alp
 	}
 
 	l.Info("getting installed bundle")
-	installedBundle, err := r.InstalledBundleGetter.GetInstalledBundle(ctx, ext)
+	// second return is status of the highest semver, if we need it
+	installedBundle, _, err := r.InstalledBundleGetter.GetInstalledBundle(ctx, ext)
 	if err != nil {
 		setInstallStatus(ext, nil)
-		// TODO: use Installed=Unknown
-		setInstalledStatusConditionFailed(ext, err.Error())
+		setInstalledStatusConditionUnknown(ext, err.Error())
 		setStatusProgressing(ext, err)
 		return ctrl.Result{}, err
 	}
 
+	// Update current install status conditions before we start trying to install something new
+	if installedBundle != nil {
+		installStatus := &ocv1alpha1.ClusterExtensionInstallStatus{
+			Bundle: *installedBundle,
+		}
+		setInstallStatus(ext, installStatus)
+		if !apimeta.IsStatusConditionTrue(ext.Status.Conditions, ocv1alpha1.TypeInstalled) {
+			setInstalledStatusConditionSuccess(ext, "Installed extension successfully")
+		}
+	} else {
+		setInstallStatus(ext, nil)
+		apimeta.RemoveStatusCondition(&ext.Status.Conditions, ocv1alpha1.TypeInstalled)
```

Comment:
Unfortunately, I don't think we can directly update the status this way. Here's the problematic scenario.
1. We start `reconcile` and we have `status.Conditions[type=Installed]` is `True`.
2. We don't find a deployed bundle, so we remove the `Installed` status condition (really, we should be setting this to `False` instead of removing it)
3. We continue on, resolve, unpack, and install successfully, resulting in us setting `status.Conditions[type=Installed]` back to `True`.

The problem is that this will cause `lastTransitionTime` to update without an actual transition of the Status. (It stays `True` in etcd the whole time).

We need to figure out a way to ensure that we only touch each condition type once in a given run of `reconcile`.

Reply:
Maybe we need a helper function that can encapsulate the update logic and then we still end up calling that helper function in all of the various error handling `if err  != nil` blocks? We'd pass `installedBundle` into it to handle setting the condition and other status fields appropriately.

Reply:
That's exactly what I did (before reading this comment)